File: lib/vn/edu/fpt/src/models/payment_configuration.dart
class PaymentConfiguration {
  const PaymentConfiguration({
    required this.id,
    required this.academicYearId,
    required this.bankCode,
    required this.bankName,
    required this.accountNumber,
    required this.accountHolder,
    required this.branch,
    required this.transferContentTemplate,
    required this.enabled,
    required this.method,
    required this.displayMode,
    required this.qrAvailable,
  });

  final int id;
  final int academicYearId;
  final String? bankCode;
  final String bankName;
  final String accountNumber;
  final String accountHolder;
  final String? branch;
  final String transferContentTemplate;
  final bool enabled;
  final String method;
  final String displayMode;
  final bool qrAvailable;

  String renderTransferContent({
    required String studentCode,
    required String academicYear,
    required String semester,
  }) {
    return transferContentTemplate
        .replaceAll('{studentCode}', studentCode)
        .replaceAll('{academicYear}', academicYear)
        .replaceAll('{semester}', semester)
        .replaceAll(RegExp(r'\s+'), ' ')
        .trim();
  }
}

File: admin-web/src/api/paymentConfiguration.ts
import { apiFetch } from './client';

export interface PaymentConfiguration {
  id: number;
  academicYearId: number;
  bankCode: string | null;
  bankName: string;
  accountNumber: string;
  accountHolder: string;
  branch: string | null;
  transferContentTemplate: string;
  enabled: boolean;
  method: 'BANK_TRANSFER';
  displayMode: 'MANUAL' | 'MANUAL_AND_QR';
  qrAvailable: boolean;
  updatedAt: string | null;
}

export interface PaymentConfigurationInput {
  bankCode: string;
  bankName: string;
  accountNumber: string;
  accountHolder: string;
  branch: string;
  transferContentTemplate: string;
  enabled: boolean;
}

export async function getPaymentConfiguration(
  academicYearId: string,
): Promise<PaymentConfiguration | null> {
  return apiFetch(`/payment-configurations/academic-years/${academicYearId}`);
}

export async function updatePaymentConfiguration(
  academicYearId: string,
  input: PaymentConfigurationInput,
): Promise<PaymentConfiguration> {
  return apiFetch(`/payment-configurations/academic-years/${academicYearId}`, {
    method: 'PUT',
    body: JSON.stringify(input),
  });
}
